src/microbe_stage/components/StrainAffected.cs

```diff
@@ -0,0 +1,34 @@
+﻿namespace Components
+{
+    using System;
+
+    /// <summary>
+    ///   Contains variables related to strain
+    /// </summary>
+    [JSONDynamicTypeAllowed]
+    public struct StrainAffected
+    {
+        /// <summary>
+        ///   True when sprinting or when strain is supposed to be otherwise generated
+        /// </summary>
+        public bool IsUnderStrain;
+
+        /// <summary>
+        ///   The current amount of strain
+        /// </summary>
+        public float CurrentStrain;
+
+        /// <summary>
+        ///   The amount of time the player has to wait before <see cref="CurrentStrain"/> sarts to fall
+        /// </summary>
+        public float StrainDecreaseCooldown;
+    }
+
+    public static class StrainAffectedHelpers
+    {
+        public static float CalculateStrainFraction(this ref StrainAffected affected)
+        {
+            return Math.Max(0, affected.CurrentStrain - Constants.CANCELED_STRAIN) / Constants.MAX_STRAIN_PER_CELL;
```

Comment:
Is the max strain per cell variable well named? I ask because for cell colonies they contain many cells. Should it perhaps be named `MAX_STRAIN_PER_ENTITY`?